### PR TITLE
address some FIXME whose associated issues were marked as closed

### PR DESCRIPTION
**Pull request by nivkner**
part of #44366

### CODE REVIEW

**Comment by michaelwoerister** [comment]
Thanks for the PR, @nivkner! I'll review as soon as I find the time (unless someone else from @rust-lang/compiler beats me to it).

File: src/tools/compiletest/src/main.rs [hunk 1]
```diff
@@ -452,11 +452,6 @@ pub fn run_tests(config: &Config) {
         _ => { /* proceed */ }
     }
 
-    // FIXME(#33435) Avoid spurious failures in codegen-units/partitioning tests.
```
Comment (estebank) [inline src/tools/compiletest/src/main.rs]: This is the only removal that gives me pause, the rest LGTM.

**Comment by nivkner** [comment]
Accidentally locked myself out of my github account (using phone app now), please put the review on hold for now. Sorry.

File: src/tools/compiletest/src/main.rs (hunk 1 above)
Comment (nivkner) [reply src/tools/compiletest/src/main.rs]: Checked the issue again. This was the fix mentioned, so reverting commit.

**Pull request merged**